src/Microsoft.OpenApi/Models/OpenApiModelFactory.cs

```diff
@@ -0,0 +1,221 @@
+        /// <returns>An OpenAPI document instance.</returns>
+        public static OpenApiDocument Parse(string input,
+                                            out OpenApiDiagnostic diagnostic,
+                                            string format = null,
+                                            OpenApiReaderSettings settings = null)
+        {
+            format ??= OpenApiConstants.Json;
+            return OpenApiReaderRegistry.GetReader(format).Parse(input, out diagnostic, settings);
+        }
+
+        /// <summary>
+        /// Reads the input string and parses it into an Open API document.
+        /// </summary>
+        /// <param name="input">The input string.</param>
+        /// <param name="version"></param>
+        /// <param name="diagnostic">The diagnostic entity containing information from the reading process.</param>
+        /// <param name="format">The Open API format</param>
+        /// <param name="settings">The OpenApi reader settings.</param>
+        /// <returns>An OpenAPI document instance.</returns>
+        public static T Parse<T>(string input,
+                                 OpenApiSpecVersion version,
+                                 out OpenApiDiagnostic diagnostic,
+                                 string format = null,
+                                 OpenApiReaderSettings settings = null) where T : IOpenApiElement
+        {
+            format ??= OpenApiConstants.Json;
+            return OpenApiReaderRegistry.GetReader(format).Parse<T>(input, version, out diagnostic, settings);
+        }
+
+        public static T Load<T>(string url, OpenApiSpecVersion version, out OpenApiDiagnostic diagnostic, OpenApiReaderSettings settings = null) where T : IOpenApiElement
+        {
+            var format = GetFormat(url);
+            return OpenApiReaderRegistry.GetReader(format).Read<T>(url, version, out diagnostic, settings);
+        }
+
+        public static T Load<T>(Stream input, OpenApiSpecVersion version, out OpenApiDiagnostic diagnostic, string format, OpenApiReaderSettings settings = null) where T : IOpenApiElement
+        {
+            format ??= OpenApiConstants.Json;
+            return OpenApiReaderRegistry.GetReader(format).Read<T>(input, version, out diagnostic, settings);
+        }
+
+        public static T Load<T>(TextReader input, OpenApiSpecVersion version, out OpenApiDiagnostic diagnostic, string format, OpenApiReaderSettings settings = null) where T : IOpenApiElement
+        {
+            format ??= OpenApiConstants.Json;
+            return OpenApiReaderRegistry.GetReader(format).Read<T>(input, version, out diagnostic, settings);
+        }
+
+
+        private static string GetContentType(string url)
+        {
+            var response = _httpClient.GetAsync(url).GetAwaiter().GetResult();
+            var contentType = response.Content.Headers.ContentType.MediaType;
+            if (contentType.EndsWith(OpenApiConstants.Json, StringComparison.OrdinalIgnoreCase))
```

Comment:
You probably want logic similar to https://github.com/microsoft/kiota-abstractions-dotnet/blob/4bff08d4ff90b24c705f71ba4d3baf89a85f31c0/src/serialization/ParseNodeFactoryRegistry.cs#L49

Using `endsWith` may bring issues in scenarios where subtypes are present.
https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html#content-type